src/main.go

```diff
@@ -1,11 +1,35 @@
 package main
 
 import (
+	"bufio"
 	"flag"
+	"os"
 	"session"
+	"strings"
+	"chat"
 )
 
 func main() {
+
 	flag.Parse()
+
+	//kick off chat loop
+	go sendChatMessageLoop()
+
 	session.SetupSession()
 }
+
+
+func sendChatMessageLoop(){
```

Comment:
Please to move these chat-specific functions inside the chat package (chat.go is fine).